### ISSUE TITLE
Sessions disappeared

### ISSUE THREAD

**Issue by geometryolife** [tmux/tmux#2326]
### Issue description
Please provide the following information:

* tmux version (tmux next-3.3)
* Platform (`Linux x86_64 `).
* $TERM inside and outside of tmux (`echo $TERM`).
Inside: xterm-256color
Outside: xterm-256color
* Logs from tmux (`tmux kill-server; tmux -vv new`).

Issue:
I often use my iPad to program on the server.
Tmux is important for me. But there is a serious problem with Tmux.
When my Termius App was interrupted unexpected and re-entered, all my sessions disappeared.
Even if I log out normally (Carl-b d), my session will still disappear when I enter again.
This makes me very distressed and hope to fix it soon.
The configuration file I use is .tmux.

**Comment by nicm** [comment]
tmux logs?

**Comment by geometryolife** [comment]
> tmux logs?
Here:
https://github.com/geometryolife/issues_files.git

**Comment by nicm** [comment]
This shows the server exiting normally, it looks to me like you pressed `C-d` in the last pane and made the shell exit.

**Comment by geometryolife** [comment]
> This shows the server exiting normally, it looks to me like you pressed `C-d` in the last pane and made the shell exit.

tmux ls
no server running on /tmp/tmux-1001/default
I think the reason why my session disappeared may be that it was stored here (/tmp).

**Comment by nicm** [comment]
You can do `pkill -USR1 tmux` to make it create the socket again.

**Comment by geometryolife** [comment]
> You can do `pkill -USR1 tmux` to make it create the socket again.

**Comment by geometryolife** [comment]
> You can do `pkill -USR1 tmux` to make it create the socket again.

This problem reappeared.
https://github.com/geometryolife/issues2.git

**Comment by nicm** [comment]
If `pkill -USR1 tmux` fixes it, then you need to find whatever is deleting your tmux socket and stop it doing so.

**Comment by geometryolife** [comment]
> If `pkill -USR1 tmux` fixes it, then you need to find whatever is deleting your tmux socket and stop it doing so.

I don't quite understand, can you find the problem from the new log?

**Comment by nicm** [comment]
Can you attach to tmux again if you do `pkill -USR1 tmux` first?

**Comment by geometryolife** [comment]
> Can you attach to tmux again if you do `pkill -USR1 tmux` first?

It doesn’t work.
I use tmux(Version—>tmux 3.0a) on another server last night, and the session has not disappeared yet.

**Comment by nicm** [comment]
When you can't connect to tmux, is it still running? What does `pgrep -lf tmux` show when `tmux attach` won't work?

**Comment by geometryolife** [comment]
> When you can't connect to tmux, is it still running? What does `pgrep -lf tmux` show when `tmux attach` won't work?

`pgrep -lf tmux`
Nothing is displayed.

Tmux can connect, but the process on the server will disappear.
When I entered the sever again, there were no resident sessions.
For example:
tmux ls
no server running on /tmp/tmux-1001/default

**Comment by nicm** [comment]
Start tmux with `tmux -vv new` then disconnect from the server, then connect again and make sure tmux has exited, then send me the tmux-server*.log file.

**Comment by geometryolife** [comment]
> Start tmux with `tmux -vv new` then disconnect from the server, then connect again and make sure tmux has exited, then send me the tmux-server*.log file.

https://github.com/geometryolife/issues2.git
In the `test` folder.

**Comment by geometryolife** [comment]
> > Start tmux with `tmux -vv new` then disconnect from the server, then connect again and make sure tmux has exited, then send me the tmux-server*.log file.
> 
> https://github.com/geometryolife/issues2.git
> In the `test` folder.

 `end.txt` file is a server log generated by a sudden interruption during normal use yesterday. Because the file is too large, only the latter is intercepted.
Maybe it helps.

**Comment by nicm** [comment]
OK do this then:
~~~~
pkill tmux
strace -tt -fostrace.out tmux -vv new
~~~~
Then reproduce and show me both the strace.out file and the tmux-server*.log file.

Please attach the entire files to this issue. You should be able to gzip them if they are big.

**Comment by geometryolife** [comment]
> OK do this then:
> 
> ```
> pkill tmux
> strace -tt -fostrace.out tmux -vv new
> ```
> 
> Then reproduce and show me both the strace.out file and the tmux-server*.log file.
> 
> Please attach the entire files to this issue. You should be able to gzip them if they are big.

https://github.com/geometryolife/issues2.git
This problem reappeared.

**Comment by nicm** [comment]
It does not look like this tmux server has exited, are you sure it is not still running? What does `pgrep -lf tmux` show?

**Comment by geometryolife** [comment]
> It does not look like this tmux server has exited, are you sure it is not still running? What does `pgrep -lf tmux` show?

Nothing.
I didn't quit tmux, I just turned off the iPad screen, went to have a meal, and looked back again, it showed that there was no conversation.(tmux next3.3)
My session on another server has been open for two days and it is still normal.(tmux 3.0a)

**Comment by nicm** [comment]
What if you run strace in nohup?
~~~~
pkill tmux
nohup strace -tt -fostrace.out tmux new
~~~~
Just show me the strace.out file.

**Comment by geometryolife** [comment]
> nohup strace -tt -fostrace.out tmux new

nohup: ignoring input and appending output to 'nohup.out'
https://github.com/geometryolife/issues2.git

**Comment by nicm** [comment]
Hmm that didn't work.

Try doing this in one terminal:

~~~~
nohup strace -tt -fostrace.out tmux -vv new -d
~~~~

Then in a different terminal do:

~~~~
tmux attach
~~~~

Then reproduce and see if the problem occurs and if so send me the log file and strace.out file.

**Comment by nicm** [comment]
That might not work either actually. Do this instead:
~~~~
tmux -vv new -d
nohup strace -tt -fostrace.out -p "$(tmux display -p #{pid})"
~~~~
Then from a different terminal:
~~~~
tmux attach
~~~~

**Comment by geometryolife** [comment]
> That might not work either actually. Do this instead:
> 
> ```
> tmux -vv new -d
> nohup strace -tt -fostrace.out -p "$(tmux display -p '#{pid}')"
> ```
> 
> Then from a different terminal:
> 
> ```
> tmux attach
> ```

Executed. And then?
Need to execute `tmux kill-server`?

**Comment by nicm** [comment]
No, you need to reproduce the problem...

**Comment by geometryolife** [comment]
> No, you need to reproduce the problem...

Ok.

**Comment by geometryolife** [comment]
> No, you need to reproduce the problem...

In the s_ser folder.
https://github.com/geometryolife/issues2.git

**Comment by nicm** [comment]
This didn't work, the strace.out file is empty. Are you sure you did this with the single quotes around `#{pid}`?
~~~~
nohup strace -tt -fostrace.out -p "$(tmux display -p '#{pid}')"
~~~~

**Comment by geometryolife** [comment]
> This didn't work, the strace.out file is empty. Are you sure you did this with the single quotes around `#{pid}`?

I copied it directly

**Comment by nicm** [comment]
Right but I edited the comment to add the `'`s, are you sure you got that version?

**Comment by geometryolife** [comment]
> Right but I edited the comment to add the `'`s, are you sure you got that version?

tmux -V
tmux next-3.3

**Comment by nicm** [comment]
No, I mean are you sure the command you copied was the last one I gave including the `'`s?

**Comment by geometryolife** [comment]
> No, I mean are you sure the command you copied was the last one I gave including the `'`s?

Yes.
tmux -vv new -d
mx➜  s_ser2 : master ✘ :✭ ᐅ  nohup strace -tt -fostrace.out -p "$(tmux display -p '#{pid}')"
nohup: ignoring input and appending output to 'nohup.out'

**Comment by nicm** [comment]
Oh yeah strace failed with:
~~~~
strace: Could not attach to process. If your uid matches the uid of the target process, check the setting of /proc/sys/kernel/yama/ptrace_scope, or try again as the root user. For more details, see /etc/sysctl.d/10-ptrace.conf: Operation not permitted
strace: attach: ptrace(PTRACE_SEIZE, 17011): Operation not permitted
~~~~
Can you try running the strace as root? First do this as your user:
~~~~
pkill tmux
tmux -vv new -d
tmux display -p '#{pid}'
~~~~
And if will show the PID. Then do as root:
~~~~
strace -tt -fostrace.out -p PID
~~~~
Where PID is the tmux PID.

Then run `tmux attach` from another terminal and reproduce and show me the strace.out file.

**Comment by geometryolife** [comment]
> Oh yeah strace failed with:
> 
> ```
> strace: Could not attach to process. If your uid matches the uid of the target process, check the setting of /proc/sys/kernel/yama/ptrace_scope, or try again as the root user. For more details, see /etc/sysctl.d/10-ptrace.conf: Operation not permitted
> strace: attach: ptrace(PTRACE_SEIZE, 17011): Operation not permitted
> ```
> 
> Can you try running the strace as root? First do this as your user:
> 
> ```
> pkill tmux
> tmux -vv new -d
> tmux display -p '#{pid}'
> ```
> 
> And if will show the PID. Then do as root:
> 
> ```
> strace -tt -fostrace.out -p PID
> ```
> 
> Where PID is the tmux PID.
> 
> Then run `tmux attach` from another terminal and reproduce and show me the strace.out file.

Executed.
Wait until the problem occurs and then send you the strace.out file

**Comment by geometryolife** [comment]
> Oh yeah strace failed with:
> 
> ```
> strace: Could not attach to process. If your uid matches the uid of the target process, check the setting of /proc/sys/kernel/yama/ptrace_scope, or try again as the root user. For more details, see /etc/sysctl.d/10-ptrace.conf: Operation not permitted
> strace: attach: ptrace(PTRACE_SEIZE, 17011): Operation not permitted
> ```
> 
> Can you try running the strace as root? First do this as your user:
> 
> ```
> pkill tmux
> tmux -vv new -d
> tmux display -p '#{pid}'
> ```
> 
> And if will show the PID. Then do as root:
> 
> ```
> strace -tt -fostrace.out -p PID
> ```
> 
> Where PID is the tmux PID.
> 
> Then run `tmux attach` from another terminal and reproduce and show me the strace.out file.

https://github.com/geometryolife/issues2.git
`s_ser2` folder.

**Comment by nicm** [comment]
Hmm I don't know what is happening because tmux is not exiting in the log and the strace does not even go up to the end of the log but stops 16 hours earlier. Are you sure you left both the tmux and the strace running?

**Comment by geometryolife** [comment]
> Hmm I don't know what is happening because tmux is not exiting in the log and the strace does not even go up to the end of the log but stops 16 hours earlier. Are you sure you left both the tmux and the strace running?

I’m sure.
I don’t know if anyone else has the same situation as me.
Maybe it was caused by my machine.

**Comment by nicm** [comment]
If you run tmux inside GNU screen and leave it attached (so detach screen not tmux), does tmux still disappear? Does screen disappear?

**Comment by geometryolife** [comment]
> If you run tmux inside GNU screen and leave it attached (so detach screen not tmux), does tmux still disappear? Does screen disappear?

screen -ls
There is a screen on:
        27703.ms        (08/03/2020 04:43:15 PM)        (Dead ???)
Remove dead screens with 'screen -wipe'.
1 Socket in /run/screen/S-Joe.

tmux ls
no server running on /tmp/tmux-1001/default

Last night, I let tmux run in screen.
Tmux has been working till now.
However, tmux still disappear just now.

**Comment by nicm** [comment]
Did you leave tmux attached inside screen? When you attach to screen again, what does it say where tmux used to be? Does it say "[detached]" or "[server exited]" or something else?

**Comment by geometryolife** [comment]
> Did you leave tmux attached inside screen? When you attach to screen again, what does it say where tmux used to be? Does it say "[detached]" or "[server exited]" or something else?

screen -ls
There is a screen on:
        1031.ms (08/05/2020 06:45:40 PM)        (Detached)
1 Socket in /run/screen/S-Joe.

mx➜  ~  ᐅ  tmux ls
0: 1 windows (created Wed Aug  5 18:46:10 2020) (attached)

**Comment by nicm** [comment]
I don't know what you are showing me.

Run tmux inside screen and leave tmux attached but detach screen instead. When tmux disappears, attach to screen and see what it says where tmux was running.

**Comment by geometryolife** [comment]
> I don't know what you are showing me.
> 
> Run tmux inside screen and leave tmux attached but detach screen instead. When tmux disappears, attach to screen and see what it says where tmux was running.

I can’t attach to screen (execute `screen -r`), when tmux disappears.
And execute ‘screen -ls’, it says `Dead???`

>>>tmux ls
no server running on /tmp/tmux-1001/default

>>>screen -r          
There is a screen on:
        1504.tmux       (08/05/2020 07:07:36 PM)        (Dead ???)
Remove dead screens with 'screen -wipe'.
There is no screen to be resumed.

>>>screen -ls
There is a screen on:
        1504.tmux       (08/05/2020 07:07:37 PM)        (Dead ???)
Remove dead screens with 'screen -wipe'.
1 Socket in /run/screen/S-Joe.

**Comment by krader1961** [comment]
@geometryolife, I've been following this issue because these types of problems intrigue me (I used to be a UNIX support engineer). I'm pretty confident the problem is not with tmux. There is something wrong with the system you're connecting to or how you are detaching from a tmux or screen session. Also, as @nicm pointed out earlier the most recent strace of the tmux server stops unexpectedly at 2020-07-31 13:07:05 UTC; just under ten hours from when it began. It does not show the tmux server terminating. Which implies that you are running strace in a fashion that leaves it prone to being killed when you disconnect the terminal session. Which might mean you are starting it directly from the terminal session that is disappearing without using `nohup`. If you are using `nohup strace ...` what does the *nohup.out* file contain?

Also, you mentioned earlier you're connecting from an iPad; i.e., a tablet. Does this problem ever occur if you attach to the tmux session via a laptop or via the KVM (keyboard/video/mouse) of the server?

**Comment by geometryolife** [comment]
> @geometryolife, I've been following this issue because these types of problems intrigue me (I used to be a UNIX support engineer). I'm pretty confident the problem is not with tmux. There is something wrong with the system you're connecting to or how you are detaching from a tmux or screen session. Also, as @nicm pointed out earlier the most recent strace of the tmux server stops unexpectedly at 2020-07-31 13:07:05 UTC; just under ten hours from when it began. It does not show the tmux server terminating. Which implies that you are running strace in a fashion that leaves it prone to being killed when you disconnect the terminal session. Which might mean you are starting it directly from the terminal session that is disappearing without using `nohup`. If you are using `nohup strace ...` what does the _nohup.out_ file contain?
> 
> Also, you mentioned earlier you're connecting from an iPad; i.e., a tablet. Does this problem ever occur if you attach to the tmux session via a laptop or via the KVM (keyboard/video/mouse) of the server?

Thanks for your reply, I have to do some tests to locate the problem.

**Comment by geometryolife** [comment]
> @geometryolife, I've been following this issue because these types of problems intrigue me (I used to be a UNIX support engineer). I'm pretty confident the problem is not with tmux. There is something wrong with the system you're connecting to or how you are detaching from a tmux or screen session. Also, as @nicm pointed out earlier the most recent strace of the tmux server stops unexpectedly at 2020-07-31 13:07:05 UTC; just under ten hours from when it began. It does not show the tmux server terminating. Which implies that you are running strace in a fashion that leaves it prone to being killed when you disconnect the terminal session. Which might mean you are starting it directly from the terminal session that is disappearing without using `nohup`. If you are using `nohup strace ...` what does the _nohup.out_ file contain?
> 
> Also, you mentioned earlier you're connecting from an iPad; i.e., a tablet. Does this problem ever occur if you attach to the tmux session via a laptop or via the KVM (keyboard/video/mouse) of the server?

 Thanks.@nicm @krader1961 
I found the cause of the problem.
It is indeed a problem with my machine and not tmux.

**Comment by krader1961** [comment]
@geometryolife, Please share the solution. I realize it might be embarrassing in a "I can't believe I did something so stupid" way, but the info may help someone in the future. :-)

**Comment by geometryolife** [comment]
> @geometryolife, Please share the solution. I realize it might be embarrassing in a "I can't believe I did something so stupid" way, but the info may help someone in the future. :-)

I uninstalled `tmux3.3next`, then installed `tmux3.1b`, the same problem still occurs.
I let tumx stay in the screen, but the screen will be killed inexplicably.
I installed tmux3.3 on another server, and I still have no problems after using it for a few days.
My friend also helped me to test and there was no problem on his machine.

From this I infer that it must be caused by my system, not tmux.
I don't know the specific reason, but tmux is normal.
